app/MMB/src/scenes/tickets/TicketDeleteButton.js

```diff
@@ -75,11 +75,34 @@ export class TicketDeleteButton extends React.Component<
     }
   };
 
+  deleteBoardingPasses = async () => {
+    try {
+      const boardingPasses = await RNFetchBlob.fs.ls(
+        this.getBoardingPassPath(),
+      );
+      boardingPasses.forEach(async boardingPass => {
+        if (boardingPass.includes(`-bookingId:${this.props.bookingId}`)) {
+          await RNFetchBlob.fs.unlink(
+            `${this.getBoardingPassPath()}/${boardingPass}`,
+          );
+        }
+      });
+    } catch (error) {
+      throw error;
+    }
+  };
+
   hasLocalFiles = async () => {
-    // TODO: Check for boardingPasses
     const existsEticket = await RNFetchBlob.fs.exists(this.getEticketPath());
+    const existsBoardingPasses = await this.hasLocalBoardingPasses();
+    this.setState({ hasLocalFiles: existsEticket || existsBoardingPasses });
+  };
 
-    this.setState({ hasLocalFiles: existsEticket });
+  hasLocalBoardingPasses = async () => {
+    const boardingPasses = await RNFetchBlob.fs.ls(this.getBoardingPassPath());
+    return boardingPasses.some(file =>
+      file.includes(`-bookingId:${this.props.bookingId}`),
```

Comment:
I guess it is unlikely that there would be pdfs which are not boarding passes and which would have a name matching this criterion in this folder. But there could be collisions. I believe it's fine, just wanted to mention it.

Reply:
since we store it in `/boardingPasses` folder structure, it would be unwise to put anything but boarding passes there.

But of course, it is not fool proof 🙈